src/goto-instrument/synthesizer/expr_enumerator.cpp

```diff
@@ -185,7 +187,7 @@ get_partitions_long(const std::size_t n, const std::size_t k)
 /// Compute all positions of ones in the bit vector `v` (1-indexed).
 std::vector<std::size_t> get_ones_pos(std::size_t v)
 {
-  const std::size_t length = sizeof(std::size_t) * 8;
+  const std::size_t length = sizeof(std::size_t) * CHAR_BIT;
```

Comment:
You are right in that care is required where `CHAR_BIT` is the right choice and where `config.ansi_c.char_width` is to be used instead. I hope to have made the right choice in all places, but of course may have made mistakes. For this particular case: as far as I understand the code, this is about the width of bytes that CBMC is being compiled for, and not whatever the verification target might be. So I claim that `CHAR_BIT` is correct. Now you are also right in that, in all likelihood, one would use a cross compiler for such a DSP, and it may well be that no platform that CBMC ever runs on has `CHAR_BIT != 8` - but I'd also like to avoid magic numbers, making it very difficult to tell whether the use of "8" is one that should be `config.ansi_c.char_width` instead, or is actually fine for it's the compile-time byte width on all relevant platforms.

Reply:
If you are convinced that we are depending on the right thing in this case, then I am happy to merge this as is.

Reply:
This specific case it is to enumerate various bit sequences, and this enumeration is done by the compiled CBMC binary.